src/DayPickerInput.js

```diff
@@ -78,21 +78,28 @@ export default class DayPickerInput extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const { month, value } = this.state;
-    const hasDifferentValue = nextProps.value !== value;
+    const { dayPickerProps: { month }, value } = this.props;
+    const {
+      dayPickerProps: { month: nextMonth },
+      value: nextValue,
+    } = nextProps;
+
+    const hasDifferentValue = nextValue !== value;
+
+    const monthExistsOnBoth = nextMonth && month;
```

Comment:
So this would become:

```js
const monthExistsOnBoth = nextDayPickerProps.month && dayPickerProps.month;
```

I'd change the variable name from `monthsExistsOnBoth` to `willUpdateDayPickerMonth` (I don't understand what `both` are or why a month should _exist_ or not :))

Reply:
very true, will update tomorrow!